### ISSUE TITLE
Incorrect Tools Path (details see description)

### ISSUE THREAD

**Issue by bettinaheim** [daveaglick/Buildalyzer#60]
Hi there, 

I was looking into using the Buildalyzer nuget package for a .net framework app (actually a multi-targeted app - framework and core) that gets called as a subprocess from within a VSIX. The necessary Microsoft.Build libraries (Microsoft.Build.dll, Microsoft.Build.Framework.dll, Microsoft.Build.Tasks.Core.dll, Microsoft.Build.Utilities.Core.dll) are packages with the VSIX (yes, I know... I couldn't yet find a way around that...). 
The projects for which to do a design time build inside the framework app are .net core style projects (netcoreapp2.0 and netstandard2.0). Unfortunately, the tools path in this particular case seems to be incorrect, leading to an "imported project not found" error (because it is looking for the Nuget.targets in the wrong place) caused by
`  <PropertyGroup>
    <NuGetRestoreTargets Condition="'$(NuGetRestoreTargets)'=='' and '$([MSBuild]::IsRunningFromVisualStudio())'=='true'">$(MSBuildToolsPath32)\..\..\..\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.targets</NuGetRestoreTargets>
    <NuGetRestoreTargets Condition="'$(NuGetRestoreTargets)'==''">$(MSBuildToolsPath)\NuGet.targets</NuGetRestoreTargets>
  </PropertyGroup> `
in the C:\Program Files\dotnet\sdk\2.1.4\Microsoft.Common.CrossTargeting.targets. 
I tried messing around with the BuildEnvironment property in the version that is here on GitHub, but ended up doing the design time build manually using a compiler directive to distinguish between the core and the framework build of the app. 
I'd still prefer using this package if possible, though, since you are probably investing quite a bit more time than me to make this work for various scenarios. So hence the question: is it currently possible to cover this scenario? If so, what would be the right usage that I apparently couldn't figure out? If not, any chance it would be possible to look into this?

Best,
Bettina

**Comment by daveaglick** [comment]
The hope is that all scenarios will be covered. That’s turning out to be pretty tricky, but I’ve been investing a lot of time in this project lately with that goal in mind. I’ve been shuffling around the API and refactoring some underlying bits recently and I _think_ I’m getting close to a major release.

Without spending too much time on the details of your problem yet, let’s see how the next release does. I’ll let you know when it’s out and then let’s circle back and see if it resolves your problem. If not we can dig further.

**Comment by daveaglick** [comment]
Buildalyzer 1.0.0 was released to NuGet last night. Can you please try it out when you get the chance and let me know the status of this issue?

FYI - the Buildalyzer 1.0.0 API has changed significantly. Revised documentation is forthcoming, but I wanted to get the release out there ASAP. Hopefully you'll be able to figure out the differences, but if not the source code of the tests is a good place to look or wait until better docs are out next week.

**Comment by dfederm** [comment]
I'm seeing this as well in v1.0.0. The workaround for me was to do this:
```
projectAnalyzer.SetGlobalProperty("MSBuildToolsPath32", msBuildProject.GetPropertyValue("MSBuildToolsPath"));
```

Granted, my scenario requires that I do a `Load` first so I already have a `Project` to query the `MSBuildToolsPath` already, but it's still pretty inconvenient.

**Comment by bettinaheim** [comment]
I finally got around to given this a go. 
There have definitively been some improvements!
Unfortunately it is not quite there yet for me, e.g. test projects won't build properly.
Running in a separate process, the easiest thing for our purposes turned out to be setting the MSBUILD_EXE_PATH before any of the Microsoft.Build.* dlls are loaded (a static constructor in MSBuild does a whole lot of magic based on that variable). That solved all issues of having to hunt down and manually set a whole bunch of global properties. 
Thanks for giving this a go & keep up the good work!

**Comment by daveaglick** [comment]
I just published Buildalyzer 2.0 which runs MSBuild out-of-process (as opposed to the old way of running it via MSBuild APIs). In general, if you can build it Buildalyzer should now be able to build it. Given how drastically different this new technique is, I'm closing all existing bug reports.

Please try the new version and if you're still having problems, please open a new issue.
